File: two-sum.cs
// 🎀 Runtime: 143 Milliseconds **⸜(｡ ˃ ᵕ ˂ )⸝♡**
// 💞 Memory : 47.10 MB ( • ̀ω•́ )✧
/*
  I have a silly idea! It is not optimized well,
  but it was fun to think about it ^-^ 

  | Num1 | Num2 | Target |
  |------|------|--------|
  | Odd  + Even = Odd    |  (3 + 2 = 5)
  | Even + Even = Even   |  (6 + 8 = 14)
  | Odd  + Odd  = Even   |  (1 + 9 = 10)
  |----------------------|
  
  If the target is an even number, the other numbers should be equal when you mod them with two!!
  The for loop will the skip the number if it is not possible to get the target! ^-^

  
*/
public class Solution {
  public int[] TwoSum(int[] nums, int target) {
    int rounds = 0; // To count how many times the loop re-started ^-^
    int index  = nums[rounds]; // Memorize a number to not use multiple loops!!
    
    bool index_is_even  = index  % 2 == 0;
    bool target_is_even = target % 2 == 0;

    for (int n = 1; n < nums.Length; n++) {
      int number = nums[n];

      bool number_is_even = number % 2 == 0;
      bool should_skip = target_is_even ^ (index_is_even == number_is_even);

      if (!should_skip && (number + index == target)) {
        return new int[] { rounds, n };
      }

      if ((n + 1) == nums.Length) { // When the loop reached to end... switch index and do setup, then do it again! ^-^
        rounds++;
        n = rounds;

        index = nums[rounds];
        index_is_even = index % 2 == 0;
      }
    }
    return null;
  }
}

File: two-sum.py
# Complexity: O(n^2) T-T
# Runtime: 1412ms

# Let's say we have an array goes from one to four and we want seven!! 
# We will use first element as an index and start with the second element!!

# [|1|, 2, 3, 4]
#       ^ 1 + 2 = 3 
### We failed go to next step :c

# [|1|, 2, 3, 4]
#          ^ 1 + 3 = 4 
### Failed again, next step please ;-;

# [|1|, 2, 3, 4]
#             ^ 1 + 4 = 5 
### Fail again, but we reached to end of the array!! So we will change the index and try again :3

# [1, |2|, 3, 4]
#          ^ 2 + 3 = 5
### We started from three because we already did!! 2 + 1

# [1, |2|, 3, 4]
#             ^ 2 + 4 = 6
### We are really close this time, change the index again!!

# [1, 2, |3|, 4]
#             ^ 3 + 4 = 7
# Yippe!! We did it!!! :3

class Solution(object):
  def twoSum(self, nums, target):
    rounds = 0 # To count how many times the loop re-started ^-^
    index  = nums[rounds] # Memorize a number to not use multiple loops!!

    step  = 1 # This will be the steps for the loop!!
    total = len(nums)
    while step < total:
      number = nums[step]

      if (number + index == target):
        return [rounds, step]
      if ((step + 1) == len(nums)):
        # When we reach to end change the index and re-start ^o^!!
        rounds+= 1
        step  = rounds
        index = nums[rounds]
      step += 1
    return []
